processing/src/main/java/io/druid/query/groupby/having/DimensionSelectorHavingSpec.java

```diff
@@ -35,15 +36,18 @@
   private static final byte STRING_SEPARATOR = (byte) 0xFF;
   private final String dimension;
   private final String value;
+  private final ExtractionFn extractionFn;
 
   @JsonCreator
   public DimensionSelectorHavingSpec(
       @JsonProperty("dimension") String dimName,
-      @JsonProperty("value") String value
+      @JsonProperty("value") String value,
+      @JsonProperty("extractionFn") ExtractionFn extractionFn
   )
   {
     this.dimension = Preconditions.checkNotNull(dimName, "Must have attribute 'dimension'");
     this.value = value;
+    this.extractionFn = extractionFn;
```

Comment:
use `IdentityExtractionFn` when it is null that will reduce the amount of if/else IMHO